src/vm/customattribute.cpp

```diff
@@ -166,8 +166,15 @@ CustomAttributeManagedValues Attribute::GetManagedCaValue(CaValue* pCaVal)
         BOOL bAllBlittableCa = arrayType != SERIALIZATION_TYPE_ENUM;
 
         if (length == (ULONG)-1)
+        {
+            GCPROTECT_BEGIN(gc)
+            {   
+                if (arrayType == SERIALIZATION_TYPE_ENUM)
```

Comment:
Ok, since this "if (arrayType == SERIALIZATION_TYPEENUM)" statement appears twice now, they should be merged and hoisted up before the "if (length == (ULONG)-1)". 

This does also require merging the two GCPROTECTED_BEGIN clauses into a single one (they can't nest.)
While it would be sufficient for your change to make it guard just the stuff in the SERIALIZATION_TYPE_SZARRAY, we might as well take this opportunity to have it guard everything after ZeroMemory(&gc, sizeof(gc)). The way the function is structured right now is pretty chancy.